<launch>

	<!-- Set compression format to png -->
	<param name="/comp/compressed/format" value="png" />
	<param name="/camera/driver/depth_registration" value="true" />	
	

	<!-- Start the camera-->
	<include file="$(find openni_launch)/launch/openni.launch" />
	<node pkg="rviz" type="rviz" name="rviz_signDetect" output="screen" required="true" />

		
	<!--<node pkg="htwAalen_signDetect" type="surfaceDetect" name="signDetect" output="screen" respawn="true" > 
		<remap from="in" to="/camera/depth_registered/image_rect_raw"/>
	</node>-->
	
	<!-- Compress Tunneling 1MB/s-->
	<node pkg="htwAalen_signDetect" type="toCompress" name="compressNode" output="screen" respawn="true" > 
 		<remap from="/comp" to="/comp" />
 		<remap from="/in" to="/camera/depth_registered/image_rect_raw" />

	</node>
	
	<!-- Uncompress -->
	<node pkg="htwAalen_signDetect" type="fromCompress" name="uncompressNode" output="screen" respawn="true" > 
 		<remap from="/uncomp" to="/uncomp2" />
 		<remap from="/in" to="/comp" />
	</node>
	
	<!-- Nodelet for creating a pointcloud out of the detector image (for debuging filters) -->
	<node pkg="nodelet" type="nodelet" name="PointCloudStandAlone"  args="manager" output="screen"/>

		<node pkg="nodelet" type="nodelet" name="points_xyzrgb_debug"
	        args="load depth_image_proc/point_cloud_xyzrgb PointCloudStandAlone --no-bond">
	    <!-- Explicit topic remappings, shouldn't need all of these -->
	    <remap from="rgb/image_rect_color"        to="/camera/rgb/image_rect_color" />
	    <remap from="rgb/camera_info"             to="/camera/rgb/camera_info" />
	    <remap from="depth_registered/image_rect" to="/uncomp2" />
	    <remap from="depth_registered/points"     to="/debug_points" />
  	</node>      	
  	
  	<!-- Depth Image Analyzer (Input) -->
	<node pkg="DepthImageAnalyzer" type="DepthImageAnalyzer" name="DepthImageAnalyzerIn" output="screen" >
		<remap from="/depthImage" to="/camera/depth_registered/image"/>
	</node>
	
	<!-- Depth Image Analyzer (Output) -->
	<node pkg="DepthImageAnalyzer" type="DepthImageAnalyzer" name="DepthImageAnalyzerOut" output="screen" >
		<remap from="/depthImage" to="/uncomp2"/>
	</node>
</launch>